### ISSUE TITLE
an issue with masked_scatter

### ISSUE THREAD

**Issue by Juan-hussain** [pytorch/pytorch#49483]
## Issue description

after some tensor manipulation the masked_scatter function output wrong values

## Code example

tgt = torch.randint(0,10,(10,10))
mask = torch.randint(0,2,(10,10)).to(torch.bool)
mask[0,7] = True 
src = -torch.ones(10,10,dtype=torch.long)
src[0]=0
src[0,7]=1
tgt.masked_scatter(mask,src)[0,7] 
gives 0 instead of 1

**Comment by ptrblck** [comment]
I think you might have misunderstood how `masked_scatter` works.
It doesn't index into `src`, but sequentially copies over `src`'s values to `tgt` as seen here:
```python
tgt = torch.zeros(10, 10).long()
mask = torch.randint(0, 2, (10, 10)).to(torch.bool)
mask[0,7] = torch.tensor([True])
src = torch.arange(1, mask.sum()+1)
res = tgt.masked_scatter(mask, src)
print(res)
> tensor([[ 1,  2,  0,  0,  3,  0,  0,  4,  0,  5],
          [ 6,  0,  7,  8,  0,  0,  0,  9,  0,  0],
          [ 0,  0,  0,  0, 10,  0,  0, 11,  0,  0],
          [ 0,  0, 12, 13, 14, 15, 16, 17, 18,  0],
          [19, 20,  0, 21,  0,  0, 22,  0,  0, 23],
          [24, 25, 26,  0,  0, 27,  0, 28, 29, 30],
          [ 0, 31,  0,  0,  0,  0,  0, 32,  0,  0],
          [ 0,  0,  0, 33,  0, 34,  0,  0,  0,  0],
          [ 0, 35,  0,  0, 36, 37, 38,  0, 39,  0],
          [40, 41, 42,  0, 43,  0, 44, 45,  0,  0]])
```

PS: I think the forum is a better place to ask questions about the usage of functions. ;)

**Comment by Juan-hussain** [comment]
Thank you very much
